scripts/get_version.mjs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import xml2js from "xml2js";
+import fs from "fs/promises";
+
+/*
+  Inputs:
+  => platform: the platform to get the current version for
+
+  Outputs:
+  => the MAJOR.MINOR.PATCH formatted version number for the given platform
+*/
+async function getVersion(platform) {
+  // xmljs can parse both plist and xml files
+  const parseFile = async filePath => await xml2js.parseStringPromise(await fs.readFile(filePath));
+  switch (platform) {
+    case "android":
+    case "browser":
+      const { widget } = await parseFile("config.xml");
+      return widget.$.version;
+    case "ios":
+    case "osx":
+      const { 
+        plist: { 
+          dict: 
+            [{ 
+              key: plistKeys, 
+              string: plistValues 
+          }] 
+        } 
+      } = await parseFile(`apple/xcode/${platform}/Outline/Outline-Info.plist`);
+      return plistValues[plistKeys.indexOf("CFBundleShortVersionString")];
+    case "windows":
+      return "1.7.0";
+    case "linux":
+      return "1.4.0";
+    case "dev":
```

Comment:
I have to handle it in this PR otherwise the builds break - I'll just do it in environment_json like you asked

Reply:
I don't understand how it would break. What needs the "dev" version?

Reply:
we pass in dev specifically as a platform in a few places, e.g. here:

https://github.com/Jigsaw-Code/outline-client/blob/1f032d026f03d173760624b478e79efbaccbaf96/src/electron/build.action.sh#L25-L26

it's ambiguous which version to use and without this `case` statement APP_VERSION ends up empty (which breaks electron)

Reply:
This just shows how crazy this currently is...

For electron/build, pass the platform as a flag.

For src/www/start.action.sh, perhaps we pass "browser" instead of dev.

Reply:
Ugh, this is worse than I thought. In the package scripts we generate the json in the build, but then we overwrite it. That's quite bad mutation. Let's pass the right platform to build and stop regenerating the json.

Reply:
Ah, I just pushed an update that sets the version number to "0.0.0-dev" if no platform is passed - if that's not ok I can do this

Reply:
You can use `uname -s` to get the OS. Should be "Windows", "Darwin" or "Linux"

Reply:
I'm not exactly sure what you mean - you're saying `get_version` should infer the platform, rather than require it to be explicitly passed in?